console/console-init/ui/src/modules/iot-device/components/AddGateways/AddGateway.tsx

```diff
@@ -52,54 +98,142 @@ export const AddGateways: React.FunctionComponent<IGatewaysProps> = ({
     setGateways([...gateways]);
   };
 
-  const alertDescription = `In AMQ IoT, any existing device directly connected to cloud can turn
-    to be a gateway by been added to other devices for their connections. Multiple gateways are supported.`;
+  const removeGatewayGroup = (group: string) => {
+    const idIndex = gatewayGroups.indexOf(group);
+    gatewayGroups.splice(idIndex, 1);
```

Comment:
add check for `idIndex>=0` for safer side.
idIndex>=0 && gatewayGroups.splice(idIndex, 1);